### PR TITLE
feat: query hca to associate dois with projects (#81)

### PR DESCRIPTION
**Pull request by hunterckx**

### CODE REVIEW

File: app/apis/azul/hca-dcp/common/entities.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  count: number;
+  format: string;
+  matrixCellCount: number;
+  totalSize: number;
+}
+
+/**
+ * Model of project value in the response from index/projects API endpoint.
+ */
+export interface ProjectResponse {
```
Comment (NoopDog) [inline app/apis/azul/hca-dcp/common/entities.ts:43]: Can this be imported from the package or is in data-browser?

**Comment by hunterckx** [comment]
Additional notes:
- We want to initialize the DOI-to-project mapping when the server starts, which is why I've enabled Next's "instrumentation" feature, since that allows running a function on server start
- Storing the projects info on the global object (as opposed to using a singleton pattern) is what makes the data accessible after the instrumentation runs and between rebuilds done by the dev server

File: app/viewModelBuilders/catalog/hca-atlas-tracker/common/viewModelBuilders.ts [hunk 2]
```diff
@@ -324,3 +325,38 @@ function getSourceDatasetPublicationColumnDef(): ColumnDef<HCAAtlasTrackerSource
     header: HCA_ATLAS_TRACKER_CATEGORY_LABEL.PUBLICATION,
   };
 }
+
+/**
+ * Get task status describing whether a source dataset is known to be in CAP.
+ * @param sourceDataset - Source dataset.
+ * @returns whether the source dataset is in CAP.
+ */
+function getSourceDatasetInCap(
```
Comment (frano-m) [inline app/viewModelBuilders/catalog/hca-atlas-tracker/common/viewModelBuilders.ts:334]: I wonder if it is worth making each of these a single function and we pass in the key `capId` or `cellxgeneCollectionId` etc to grab the value, and then return DONE or REQUIRED?

Or, do you anticipate the returned values could become more complex and unable to be generalised?

File: app/utils/hca-projects.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  }
+  return projectIdsByDoi;
+}
+
+/**
+ * Fetch current default HCA catalog name.
+ * @returns catalog name.
+ */
+async function getLatestCatalog(): Promise<string> {
```
Comment (frano-m) [inline app/utils/hca-projects.ts:86]: @hunterckx are we able to utilise `fetchCatalog` from [findable](https://github.com/clevercanary/data-explorer/blob/main/packages/data-explorer-ui/src/entity/api/service.ts#L86)?

**Review by frano-m — APPROVED** [review]
Excellent work @hunterckx thank you ⭐ 🚀

**Pull request merged**